README.md

```diff
@@ -1,2 +1,166 @@
-# roadmap
-Master roadmap for Project Jupyter
+# Project Jupyter Roadmap
+
+The following is a very rough outline of the roadmap for Project Jupyter and IPython.
+Historically, we have published major releases every 6-9 months. With our project now
+split across many subprojects, we expect to release subprojects separately and more
+often.
+
+This roadmap only includes mainline subproject under the Jupyter
+(https://github.com/jupyter) and IPython (https://github.com/ipython) GitHub orgs. We
+also have a number of newer, experimental subprojects in the Jupyter Incubator
+(https://github.com/jupyter-incubator) organization.
+
+In this roadmap, we refer to the future major releases as follows:
+
+* Next (the next major release)
+* Next+1 (the major release after that)
+* Future (any release beyond the Next+1)
+
+This reflects the inherant uncertainty in releases that are more than about 1 year away. In
+some cases, if progress is made more quickly, features may be moved to earlier releases.
+
+## Documentation
+
+We are choosing to list documentation as a top-level section in our roadmap to acknowledge
+both the importance of it and the painful state of our existing documentation. Because our
+documentation spans all subprojects, each of which have their own release cycles and
+versioning, the documentation roadmap doesn't have concrete version numbers associated with
+each milestone.
+
+### Next 
+
+* Hire full time and part time staff who will focus on improving our documentation.
+* Research existing tools for writing documentation (Readthedocs, Sphinx, Notebooks, etc.)
+  and evaluate which tools we use to write documentation.
+* Develop a project wide outline of our existing and proposed documentation.
+* Perform observational and survey based user studies of our documentation to guide its
+  design and development.
+* Explore and research the questions around how to maintain documentation across multiple
+  subprojects. Do we centralize all or some of our documentation? Do we keep it in separate
+  subproject repos?
+* Build publicly accessible API documentation on all commits for all JavaScript and
+  TypeScript repositories.
+
```

Comment:
* Encode in our project principles and community process documentation as a core activity, that is part of the everyday process of development, and adopt this in an integral fashion throughout the project for all project members.